### PR TITLE
chore(api,worker): Just deploy Novu Cloud to staging

### PR DESCRIPTION
**Pull request by SokratisVidros**
### What changed? Why was the change needed?

When merging to next branch, there is no need to build a community docker image.

### CODE REVIEW

File: .github/workflows/dev-deploy-api.yml [hunk 1]
```diff
@@ -41,59 +40,51 @@ jobs:
...
       deployments: write
       id-token: write
-    strategy:
-      matrix:
-        name: ['novu/api-ee', 'novu/api']
     steps:
       - uses: actions/checkout@v4
         with:
-          submodules: ${{ contains (matrix.name,'-ee') }}
+          submodules: true
```
Comment (scopsy) [inline .github/workflows/dev-deploy-api.yml:46]: @SokratisVidros just to clarify, the goal with this PR is to have a separate pipeline for Community Docker Images?

Comment (SokratisVidros) [reply .github/workflows/dev-deploy-api.yml:46]: Yes. See [here](https://github.com/novuhq/novu/blob/5edb4dabe8c3522da722dd68f8033dd3f97348cb/.github/workflows/prepare-self-hosted-release.yml#L1). The `next` branch is our trunk, and it should only be deployed to staging ASAP for testing.

**Review by rifont — APPROVED** [review]
🚀

**Comment by SokratisVidros** [comment]
For the record, this change should also be applied to the remaining services including ws, webhooks and inbound.

**Pull request merged**